### ISSUE TITLE
Multiremote setValue malformed type when using Appium combined with Chromedriver.  

### ISSUE THREAD

**Issue by akume** [webdriverio/webdriverio#4069]
**Environment (please complete the following information):**
 - **WebdriverIO version:** "webdriverio": "^5.10.0",
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running:** sync
 - **Node.js version:** 8.11.3
 - **NPM version:** [e.g. 5.8.0] 6.9.0
 - **Browser name and version:** Chrome Version 74.0.3729.169 
 - **Platform name and version:** OSX 10.14.5
 - **Additional wdio packages used (if applicable):**  
    "@wdio/allure-reporter": "^5.9.3",
    "@wdio/appium-service": "^5.9.3",
    "@wdio/cli": "^5.10.0",
    "@wdio/dot-reporter": "^5.9.3",
    "@wdio/local-runner": "^5.10.0",
    "@wdio/mocha-framework": "^5.9.4",
    "@wdio/sauce-service": "^5.9.3",
    "@wdio/selenium-standalone-service": "^5.9.3",
    "@wdio/spec-reporter": "^5.9.3",
    "@wdio/sync": "^5.10.0",

**Config of WebdriverIO**

```
const chai = require('chai'); 
const { join } = require('path');

exports.config = {
  specs: ['./test/e2e/specs/*.js'],
  execArgv: ['--inspect-brk=127.0.0.1:5859'],
  logLevel: 'trace',
  bail: 0,
  waitforTimeout: 10000,
  framework: 'mocha',
  reporters: ['dot', 'spec'],
  sync: true,
  coloredLogs: true,
  screenshotPath: './errorShots/',
  connectionRetryTimeout: 90000,
  connectionRetryCount: 3,
  mochaOpts: {
    ui: 'bdd',
    timeout: 60000,
    compilers: ['js:babel-register']
  },
  services: ['appium'],
  appium: {
    args: {
    }
  },
  deprecationWarnings: true,
  capabilities: {
    appiumRuntime: {
      port: 4723,
      capabilities: {
        // The defaults you need to have in your config
        automationName: 'Appium',
        platformName: 'Android',
        platformVersion: '8.1',
        deviceName: 'Android',
        app: join(process.cwd(), '../../apps/1.3.2d.apk'),
        newCommandTimeout: 240,
        udid: "123456",
        orientation: 'PORTRAIT',
        // Read the reset strategies very well, they differ per platform, see
        // http://appium.io/docs/en/writing-running-appium/other/reset-strategies/
        noReset: false
      }
    },
    chromeIde: {
      path: '/',
      port: 4444,
      capabilities: {
        browserName: 'chrome'
      }
    }
  },
  
  before: async (capabilities, specs) => {
    var chai = require('chai');
    global.expect = chai.expect;
    global.assert = chai.assert;

    if (appiumRuntime && appiumRuntime.getContexts) {
      const contextNames = await appiumRuntime.getContexts();
      for (let i = 0; i <= contextNames.length; i++) {
        if (contextNames[i] === 'WEBVIEW_com.au.app') {
          await appiumRuntime.switchContext(contextNames[i]);
          break;
        }
      }
    }
  }
}
```

sample login test sent only to the chromeBrowser
```
describe('Remote Expert', () => {
  it('should navigate to assist', (done) => {
    chromeIde.url('http://localhost:8080');
    chromeIde.$('#login-username').setValue('da@mailinator.com');
    chromeIde.$('#login-password').setValue('password');
    chromeIde.$('#login-submit').click();
    assert(true, 'its true')
  });
});
```

**Describe the bug**
When I setup the following test and attempt to use setValue or addValue,  I get: "Malformed type for "text" parameter of command elementSendKeys Expected: string Actual: object"

If I comment out everything related to appium, the above test works fine.  

**Expected behavior**
The instructions sent to the chrome browser should work and not be affected by having an appium driver present since these instructions are specific to Chrome.  

**Log**
 

> "spec" Reporter:
> ------------------------------------------------------------------
> [MultiremoteBrowser #0-0] Spec: /Users/akume/Documents/Work/Au/code/ide/test/e2e/specs/remoteexpert.spec.js
> [MultiremoteBrowser #0-0] Running: undefined
> [MultiremoteBrowser #0-0]
> [MultiremoteBrowser #0-0] Remote Expert
> [MultiremoteBrowser #0-0]    ✖ should navigate to assist
> [MultiremoteBrowser #0-0]
> [MultiremoteBrowser #0-0] 1 failing (2m 15.1s)
> [MultiremoteBrowser #0-0]
> [MultiremoteBrowser #0-0] 1) Remote Expert should navigate to assist
> [MultiremoteBrowser #0-0] Malformed type for "text" parameter of command elementSendKeys
> Expected: string
> Actual: object
> 
> For more info see https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol#sessionsessionidelementidvalue
> 
> [MultiremoteBrowser #0-0] Error: Malformed type for "text" parameter of command elementSendKeys
> [MultiremoteBrowser #0-0] Expected: string
> [MultiremoteBrowser #0-0] Actual: object
> [MultiremoteBrowser #0-0] For more info see https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol#sessionsessionidelementidvalue
> [MultiremoteBrowser #0-0]     at Element.<anonymous> (/Users/akume/Documents/Work/Au/code/ide/node_modules/webdriver/build/command.js:66:15)
> [MultiremoteBrowser #0-0]     at <anonymous>
> [MultiremoteBrowser #0-0]     at Context.<anonymous> (/Users/akume/Documents/Work/Au/code/ide/test/e2e/specs/remoteexpert.spec.js:18:36)
> 
> 
> Spec Files:      0 passed, 1 failed, 1 total (100% completed) in 00:02:30

**Comment by christian-bromann** [comment]
Can you provide some logs?

**Comment by akume** [comment]
> AMBP:e2e akume$ cd /Users/akume/Documents/Work/Augmentir/code/e2e ; env DEBUG=1 /Users/akume/Documents/Work/Augmentir/code/e2e/node_modules/.bin/wdio ./config/wdio.conf.js --spec ./specs/remoteexpert.spec.js 
> 2019-06-10T15:52:58.153Z WARN @wdio/config:ConfigParser: pattern ./test/e2e/specs/*.js did not match any file
> 
> Execution of 1 spec files started at 2019-06-10T15:52:58.191Z
> 
> 2019-06-10T15:52:58.193Z DEBUG @wdio/utils:initialiseServices: initialise wdio service "appium"
> 2019-06-10T15:52:58.203Z INFO @wdio/cli:Launcher: Run onPrepare hook
> 2019-06-10T15:52:58.205Z DEBUG @wdio/appium-service: Will spawn Appium process: /Users/akume/Documents/Work/Augmentir/code/e2e/node_modules/appium/build/lib/main.js 
> 2019-06-10T15:52:59.872Z DEBUG @wdio/appium-service: Appium started with ID: 56700
> 2019-06-10T15:52:59.876Z INFO @wdio/local-runner: Start worker 0-0 with arg: ./config/wdio.conf.js,--spec,./specs/remoteexpert.spec.js
> [0-0] RUNNING in MultiRemote - /specs/remoteexpert.spec.js
> [0-0] 2019-06-10T15:53:00.519Z INFO @wdio/local-runner: Run worker command: run
> [0-0] 2019-06-10T15:53:00.566Z WARN @wdio/config:ConfigParser: pattern ./test/e2e/specs/*.js did not match any file
> [0-0] 2019-06-10T15:53:00.568Z DEBUG @wdio/utils:initialiseServices: initialise wdio service "appium"
> [0-0] 2019-06-10T15:53:00.614Z DEBUG @wdio/local-runner:utils: init multiremote session
> [0-0] 2019-06-10T15:53:00.679Z INFO webdriver: [POST] http://127.0.0.1:4723/wd/hub/session
> [0-0] 2019-06-10T15:53:00.680Z INFO webdriver: DATA { capabilities:
>    { alwaysMatch:
>       { automationName: 'Appium',
>         platformName: 'Android',
>         platformVersion: '8.1',
>         deviceName: 'Da Android',
>         app:
>          '/Users/akume/Documents/Work/Augmentir/apps/android-debug-1.3.2d.apk',
>         newCommandTimeout: 240,
>         udid: '96d9aae2',
>         orientation: 'PORTRAIT',
>         noReset: false },
>      firstMatch: [ {} ] },
>   desiredCapabilities:
>    { automationName: 'Appium',
>      platformName: 'Android',
>      platformVersion: '8.1',
>      deviceName: 'Da Android',
>      app:
>       '/Users/akume/Documents/Work/Augmentir/apps/android-debug-1.3.2d.apk',
>      newCommandTimeout: 240,
>      udid: '96d9aae2',
>      orientation: 'PORTRAIT',
>      noReset: false } }
> [0-0] 2019-06-10T15:53:00.692Z INFO webdriver: [POST] http://127.0.0.1:4444/session
> [0-0] 2019-06-10T15:53:00.693Z INFO webdriver: DATA { capabilities:
>    { alwaysMatch: { browserName: 'chrome' }, firstMatch: [ {} ] },
>   desiredCapabilities: { browserName: 'chrome' } }
> [0-0] 2019-06-10T15:53:09.322Z INFO webdriver: COMMAND getContexts()
> [0-0] 2019-06-10T15:53:09.324Z INFO webdriver: [GET] http://127.0.0.1:4723/wd/hub/session/494fb7fa-4d7b-4b0a-8adb-a963186f9da6/contexts
> [0-0] 2019-06-10T15:53:09.762Z INFO webdriver: RESULT [ 'NATIVE_APP', 'WEBVIEW_chrome', 'WEBVIEW_com.augmentir.app' ]
> [0-0] 2019-06-10T15:53:09.765Z INFO webdriver: COMMAND switchContext("WEBVIEW_com.augmentir.app")
> [0-0] 2019-06-10T15:53:09.766Z INFO webdriver: [POST] http://127.0.0.1:4723/wd/hub/session/494fb7fa-4d7b-4b0a-8adb-a963186f9da6/context
> [0-0] 2019-06-10T15:53:09.768Z INFO webdriver: DATA { name: 'WEBVIEW_com.augmentir.app' }
> [0-0] 2019-06-10T15:53:14.385Z INFO webdriver: COMMAND navigateTo("http://localhost:8080/")
> [0-0] 2019-06-10T15:53:14.386Z INFO webdriver: [POST] http://127.0.0.1:4444/session/23880987e9ad137bf1c86d75ca282416/url
> [0-0] 2019-06-10T15:53:14.387Z INFO webdriver: DATA { url: 'http://localhost:8080/' }
> [0-0] 2019-06-10T15:53:16.168Z INFO webdriver: COMMAND findElement("css selector", "#login-username")
> [0-0] 2019-06-10T15:53:16.170Z INFO webdriver: [POST] http://127.0.0.1:4444/session/23880987e9ad137bf1c86d75ca282416/element
> [0-0] 2019-06-10T15:53:16.171Z INFO webdriver: DATA { using: 'css selector', value: '#login-username' }
> [0-0] 2019-06-10T15:53:16.197Z INFO webdriver: RESULT { ELEMENT: '0.06150234630065987-1' }
> [0-0] 2019-06-10T15:53:16.290Z INFO webdriver: COMMAND elementClear("0.06150234630065987-1")
> [0-0] 2019-06-10T15:53:16.291Z INFO webdriver: [POST] http://127.0.0.1:4444/session/23880987e9ad137bf1c86d75ca282416/element/0.06150234630065987-1/clear
> [0-0] Error in "Remote Expert should navigate to assist"
> Malformed type for "text" parameter of command elementSendKeys
> Expected: string
> Actual: object
> 
> For more info see https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol#sessionsessionidelementidvalue
> 
> [0-0] 2019-06-10T15:53:16.338Z INFO webdriver: COMMAND deleteSession()
> [0-0] 2019-06-10T15:53:16.339Z INFO webdriver: [DELETE] http://127.0.0.1:4444/session/23880987e9ad137bf1c86d75ca282416
> [0-0] 2019-06-10T15:53:16.341Z INFO webdriver: COMMAND deleteSession()
> [0-0] 2019-06-10T15:53:16.342Z INFO webdriver: [DELETE] http://127.0.0.1:4723/wd/hub/session/494fb7fa-4d7b-4b0a-8adb-a963186f9da6
> 2019-06-10T15:53:18.127Z DEBUG @wdio/local-runner: Runner 0-0 finished with exit code null
> [0-0] FAILED in MultiRemote - /specs/remoteexpert.spec.js
> 2019-06-10T15:53:18.128Z INFO @wdio/cli:Launcher: Run onComplete hook
> 2019-06-10T15:53:18.129Z DEBUG @wdio/appium-service: Appium (pid: 56699) killed
>

**Comment by christian-bromann** [comment]
You need to provide at least the full Appium logs

**Comment by akume** [comment]
Here is the appium logs

[debug-log.txt](https://github.com/webdriverio/webdriverio/files/3273106/debug-log.txt)

**Comment by christian-bromann** [comment]
> Here is the appium logs

This log doesn't contain any errors

**Comment by akume** [comment]
the issue isnt on the appium side.  the problem is that when I define appium in addition to a chromedriver in a multiremote scenario, chromedriver fails to setValue or addValue.  If i just use firefox and chrome, this problem does not occur.

**Comment by christian-bromann** [comment]
I guess this has something to do with Appium and WebdriverIO not applying the right protocol to the session. I don't have the time to look into this so if you find anything, let us know.

**Comment by CrispusDH** [comment]
@akume could you update `webdriverio` to the last version and check again?

**Comment by ericzered** [comment]
Hello, I've got the exact same problem, with the last 5.10.1 webdriverio version and the very last appium 1.13. (I had the probleme too with the 5.9.1 webdriverio version).
I see that if I change this line : 
https://github.com/webdriverio/webdriverio/blob/master/packages/webdriverio/src/commands/element/addValue.js#L32
to put this line instead :
https://github.com/webdriverio/webdriverio/blob/master/packages/webdriverio/src/commands/element/addValue.js#L37
it's ok on Chrome Desktop when android app browser is running in the same time... but it's not working anymore when I have only Desktop browser.
So I didn't find a way to make it works each time
Thanks !

**Comment by mgrybyk** [comment]
I think this is not an issue of appium or wdio, try changing chrome mode to non w3c

**Comment by ericzered** [comment]
I'ce tried to put capabilities chrome like this : 
chrome: {
    capabilities: {
      browserName: 'chrome',
      'goog:chromeOptions': {
        'w3c': false,
        args: ['--no-first-run'],
      }
    },
  },
but still same error

**Comment by ericzered** [comment]
But I can add that if I run Firefox + Android, I don't have this problem. It's only happens on Chrome + Android. Maybe I don't set non w3 in a wrong way

**Comment by akume** [comment]
the problem is with webdriverio/packages/webdriver/src/utils.js line 139 and 144 respectively.  The losash.merge is mutating the protocol.  There should be an empty object at the beginning of this command.  

```
export function getPrototype ({ isW3C, isChrome, isMobile, isSauce }) {
    const prototype = {}
    const ProtocolCommands = merge(
        /**
         * if mobile apply JSONWire and WebDriver protocol because
         * some legacy JSONWire commands are still used in Appium
         * (e.g. set/get geolocation)
         */
        isMobile
            ? **merge({}, JsonWProtocol, WebDriverProtocol)**
            : isW3C ? WebDriverProtocol : JsonWProtocol,
        /**
         * only apply mobile protocol if session is actually for mobile
         */
        isMobile ? **merge({}, MJsonWProtocol, AppiumProtocol)** : {},
        /**
         * only apply special Chrome commands if session is using Chrome
         */
        isChrome ? ChromiumProtocol : {},
        /**
         * only Sauce Labs specific vendor commands
         */
        isSauce ? SauceLabsProtocol : {}
    )

    for (const [endpoint, methods] of Object.entries(ProtocolCommands)) {
        for (const [method, commandData] of Object.entries(methods)) {
            prototype[commandData.command] = { value: command(method, endpoint, commandData) }
        }
    }

    return prototype
}

```

**Comment by akume** [comment]
https://github.com/webdriverio/webdriverio/pull/4077

**Comment by christian-bromann** [comment]
merged #4077

**Comment by ericzered** [comment]
Thanks it's ok now with this PR !
